Второе домашнее задание

NBronina opened the pull request:
Не поняла, на что ругается http://jslint.com/

NBronina committed:
Создание объекта, описывающего события в календаре

Создание объекта. Есть ошибки при проверке в http://jslint.com/

azproduction commented:
На `Unexpected 'typeof'. Use '===' to compare directly with undefined.` можно не обращать вимание - эту проверку никак не отключить.

azproduction commented on dz2.js:
```diff
@@ -0,0 +1,48 @@
+ * @return {Object}
+ */
+/*jslint browser: true, devel: true */
+function Event(name, start, end, participants, organizer, place, info, reminder, type) {
```
Функция имеет слишком много параметров подумай как можно уменьшить их число.

azproduction commented on dz2.js:
```diff
@@ -0,0 +1,48 @@
+ * @param {String} place                                   Место встречи
+ * @param {String} info                                    Дополнительная информация о встрече
+ * @param {Number|Date} reminder                           Дата, когда напомнить о встрече
+ * @param {String} type["Работа", "Отдых", "Дела"]         Тип встречи, пользователю будет предложено выбрать из списка
```
Выравнивание имен переменных

```
 * @param {Number|Date} end                                Конец события
 * @param {Object}      participants                       Список участников и их контакты
```

azproduction commented on dz2.js:
```diff
@@ -0,0 +1,48 @@
+        "organizer": organizer || {},
+        "place": place || {},
+        "info": info || {},
+        "reminder": reminder || function WorkData(start) {
```
Тип неконсистентрый `{Number|Date} reminder`, а фактически может быть Function. Лучше лишний раз функции не порождать.

NBronina committed:
исправлена ошибка, добавлены поля

изменен тип поля "reminder", добавлено поле "party" для подтверждения
участия в событии
